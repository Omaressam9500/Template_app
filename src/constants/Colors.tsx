export const lightTheme = {
    backgroundColor: '#F9F9FF',
    tabBarBackground: "#FFFFFF",
    textColor: '#262626',
    headerTxtColor: "#10C7E3",
    screenBackgroundColor: '#F9F9FF',
    white : '#FFFFFF',
    black : '#000000',

};

export const darkTheme = {
    backgroundColor: '#262626',
    tabBarBackground: "#4A4A4A",
    textColor: "#FFFFFF",
    headerTxtColor: "#10C7E3",
    screenBackgroundColor: "#373737",
    black : '#FFFFFF',
    white : '#1F1F1F',


};
