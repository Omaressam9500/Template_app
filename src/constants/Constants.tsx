export const Constants = {
    API_KEY: "b60c9667da79431481cdc1c8f56a5912",
    LANG_CODE:"LANG_CODE"
}