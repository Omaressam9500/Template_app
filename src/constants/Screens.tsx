export const Screens = {
    HOME: "HOME",
    NEWSFEED:"NEWSFEED",
    ARTICLES_DETAILS:"ARTICLES_DETAILS",
    SETTINGS: "SETTINGS",
}