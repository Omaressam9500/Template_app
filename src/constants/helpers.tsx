import moment from "moment";

import AsyncStorage from "@react-native-async-storage/async-storage";

export const formatDate = (dateString: string) => {

    return moment(dateString).format('YYYY/MM/DD hh:mm a');

};
export const saveItem = async (key: string, data: any) => {
    try {
        await AsyncStorage.setItem(key, JSON.stringify(data));
        return true;
    } catch (error) {
    }
    return false;
};

export const getItem = async (key: string) => {
    try {
        const retrievedItem: any = await AsyncStorage.getItem(key);
        return JSON.parse(retrievedItem);
    } catch (error) {
    }
    return null;
};


export const removeItem = async (key: string) => {
    try {
        await AsyncStorage.removeItem(key);
        return true;
    } catch (error) {
    }
    return false;
};