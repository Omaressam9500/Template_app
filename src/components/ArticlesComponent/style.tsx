import { StyleSheet } from 'react-native';
import { Pixel } from '../../constants/styleConstatnts';
import { useTheme } from '../../context/ThemeProvider';
export const styles = () => {
    const { theme } = useTheme();

    return StyleSheet.create({

        card: {
            backgroundColor: theme.backgroundColor,
            borderRadius: 8,
            shadowColor: '#000',
            shadowOffset: { width: 0, height: 2 },
            shadowOpacity: 0.2,
            shadowRadius: 4,
            elevation: 3,
            marginBottom: 20,
        },
        imageStyle: {
            width: '100%',
            height: 200,
            borderTopLeftRadius: 8,
            borderTopRightRadius: 8,
    
        },
        content: {
            padding: 10,
        },
        titleStyle: {
            fontSize: 18,
            fontWeight: 'bold',
            marginBottom: 5,
            alignSelf: 'flex-start',
            color:theme.textColor
        },
        descriptionStyle: {
            fontSize: 16,
            alignSelf: 'flex-start',
            color:theme.textColor
        },
    })
}