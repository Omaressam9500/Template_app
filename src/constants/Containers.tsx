import React, { FC, forwardRef } from "react";
import {
  Animated,
  ScrollView,
  ScrollViewProps,
  StatusBar,
  StyleProp,
  View,
  ViewStyle,
} from "react-native";
import { Colors, Pixel, ScreenOptions } from "../constants/styleConstatnts";
import { useTheme } from "../context/ThemeProvider";

interface containerProps {
  children?: JSX.Element[] | JSX.Element;
  style?: StyleProp<ViewStyle>;
  noPadding?: boolean;

}




export const Container: FC<containerProps> = ({ children, style, noPadding }) => {
const { theme, } = useTheme();

  return (
    <View
      style={[{
        flex: 1, backgroundColor: theme.screenBackgroundColor,
        paddingTop: noPadding ? 0 : ScreenOptions.StatusBarHeight,
      }, style]}>
      <StatusBar
        // translucent={true}
        backgroundColor={"black"}
        // barStyle="dark-content"
      />
      {children}
    </View>
  );
};


