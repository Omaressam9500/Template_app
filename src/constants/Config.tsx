import axios from "axios";

import { I18nManager } from "react-native";
import RNRestart from "react-native-restart";
import { Constants } from "./Constants";
import { saveItem } from "./helpers";

const { isRTL, forceRTL, allowRTL, swapLeftAndRightInRTL } = I18nManager;

export const headers = {
    Accept: "application/json",
    "Content-Type": "application/json",
};
export enum SERVER {
    PROD = "https://newsapi.org/v2/",
}

export const axiosAPI = axios.create({
    baseURL: SERVER.PROD,
    headers: headers,
});

axiosAPI.interceptors.response.use(function (response) {
    return response;

}, async function (error) {
    if (401 === error?.response?.status) {

    } else {
        return Promise.reject(error);
    }
})

export const ChangeLanguageHandler = async (
    code: string,
) => {
    await saveItem(Constants.LANG_CODE, code);
    allowRTL(code === "ar");
    forceRTL(code === "ar");
    swapLeftAndRightInRTL(code === "ar");
    setTimeout(() => {
        RNRestart.Restart();
    }, 300);

};