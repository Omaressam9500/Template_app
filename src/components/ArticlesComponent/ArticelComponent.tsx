import React, { FC } from "react";
import { StyleSheet, Text, View, Image, TouchableOpacity, } from "react-native";

import { useNavigation } from "@react-navigation/native";
import { Pixel } from "../../constants/styleConstatnts";
import { ImageComponent } from "../ImageComponent";
import { Screens } from "../../constants/Screens";
import { styles } from "./style";

interface IArticelComponent {
    post: IArticle
}
export const ArticelComponent: FC<IArticelComponent> = ({
    post
}) => {
    const { navigate } = useNavigation();
    const { author, urlToImage, title } = post;
    const { card, imageStyle, content, titleStyle, descriptionStyle } = styles()
    return (
        <TouchableOpacity
        testID="CardTestId"
         style={card}
            onPress={() => {
                navigate(Screens.ARTICLES_DETAILS, {
                    post: post
                })
            }}>
            <ImageComponent
                url={urlToImage}
                style={imageStyle}
                resizeMode={"cover"}
            />

            <View style={content}>
                <Text testID={"AuthorTestId"}
                    style={titleStyle} numberOfLines={2} ellipsizeMode="tail">{author}</Text>
                <Text testID={"TitleTestId"}
                    style={descriptionStyle} numberOfLines={2} ellipsizeMode="tail">{title}</Text>

            </View>

        </TouchableOpacity>


    )
}
