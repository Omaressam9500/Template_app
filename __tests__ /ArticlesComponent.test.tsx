import 'react-native';
import React from 'react';

import { it, expect, describe, jest } from '@jest/globals';
import { render, fireEvent } from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import { ArticelComponent } from '../src/components/ArticlesComponent/ArticelComponent';
import { Screens } from '../src/constants/Screens';

const mockPost = {
    author: 'Omar essam',
    urlToImage: 'https://example.com/image.jpg',
    title: 'Example Article',
    describe
  };
  const mockedNavigation = jest.fn();
  
  jest.mock('@react-navigation/native', () => {
    return {
      useNavigation: () => ({
        navigate: mockedNavigation,
      }),
    };
  });
describe('ArticelComponent', () => {

    it('Render ComponentCorrect', () => {
      const tree = renderer.create(
          <ArticelComponent post={mockPost} />,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('renders author and title correctly', () => {
  
      const { getByTestId, getByText } = render(<ArticelComponent post={mockPost} />);
  
      const authorElement = getByTestId('AuthorTestId');
      const titleElement = getByTestId('TitleTestId');
  
      expect(authorElement.props.children).toEqual('Omar essam');
      expect(titleElement.props.children).toEqual('Example Article');
    });
    it('calls navigation function when pressed', () => {
  
  
      const { getByTestId } = render(<ArticelComponent post={mockPost} />);
      const cardElement = getByTestId('CardTestId');
  
      fireEvent.press(cardElement);
  
      expect(mockedNavigation).toHaveBeenCalledWith(Screens.ARTICLES_DETAILS, { post: mockPost });
    });
  
  });