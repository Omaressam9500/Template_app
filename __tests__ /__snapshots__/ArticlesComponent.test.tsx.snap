// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticelComponent Render ComponentCorrect 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#F9F9FF",
      "borderRadius": 8,
      "elevation": 3,
      "marginBottom": 20,
      "opacity": 1,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 4,
    }
  }
  testID="CardTestId"
>
  <View>
    <View
      style={
        [
          {
            "overflow": "hidden",
          },
          [
            {
              "height": 50,
              "marginBottom": 10,
              "width": 50,
            },
            {
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "height": 200,
              "width": "100%",
            },
          ],
        ]
      }
    >
      <FastImageView
        defaultSource={null}
        resizeMode="cover"
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "padding": 10,
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        {
          "alignSelf": "flex-start",
          "color": "#262626",
          "fontSize": 18,
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }
      testID="AuthorTestId"
    >
      Omar essam
    </Text>
    <Text
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        {
          "alignSelf": "flex-start",
          "color": "#262626",
          "fontSize": 16,
        }
      }
      testID="TitleTestId"
    >
      Example Article
    </Text>
  </View>
</View>
`;
